src/index.js

```diff
@@ -40,40 +41,72 @@ function glamorous(comp, {rootEl, displayName} = {}) {
      *   merged predicably (with no regard to specificity)
      * @return {ReactElement} React.createElement
      */
-    function GlamorousComponent(props) {
-      const {className, ...rest} = props
-      const {toForward, cssOverrides} = splitProps(rest, GlamorousComponent)
-
-      // create className to apply
-      const mappedArgs = GlamorousComponent.styles.map(mapToPropsCall)
-      const {
-        glamorStyles: parentGlamorStyles,
-        glamorlessClassName,
-      } = extractGlamorStyles(className)
-      const glamorClassName = css(
-        ...mappedArgs,
-        ...parentGlamorStyles,
-        cssOverrides,
-      ).toString()
-      const fullClassName = joinClasses(glamorlessClassName, glamorClassName)
-
-      return React.createElement(GlamorousComponent.comp, {
-        className: fullClassName,
-        ...toForward,
-      })
-
-      function mapToPropsCall(glamorRules) {
-        if (typeof glamorRules === 'function') {
-          return glamorRules(props)
+    class GlamorousComponent extends Component {
+      constructor(props, context) {
```

Comment:
If we use public class fields, then we wont even need this constructor. Could just be:

```javascript
class GlamorousComponent extends Component {
  state = {theme: null}
  setTheme = () => this.setState({theme})
  // etc.
}
```